py/selenium/webdriver/firefox/firefox_profile.py

```diff
@@ -327,6 +327,16 @@ def get_text(element):
                     rc.append(node.data)
             return ''.join(rc).strip()
 
+        def parse_manifest_json(content):
+            """Extracts the details from the contents of a WebExtensions `manifest.json` file."""
+            manifest = json.loads(content)
+            return {
+                'id': manifest['applications']['gecko']['id'],
```

Comment:
The id is not mandatory for web extensions. Could you update this to support a web extension without an id?

Reply:
Here is how the ruby bindings solve it for extensions without an id: https://github.com/SeleniumHQ/selenium/blob/07a18746ff756e90fd79ef253a328bd7dfa9e6dc/rb/lib/selenium/webdriver/firefox/extension.rb#L85-L91

As seen above, the ruby bindings doesn't even look for applications.gecko.id (an issue worth reporting?)

Reply:
The Java bindings does the same as the Ruby bindings:
https://github.com/SeleniumHQ/selenium/blob/ba32c5bc83272ff0e111fd4dccc40435700c6225/java/client/src/org/openqa/selenium/firefox/internal/FileExtension.java#L152-L155

The JavaScript bindings throw an error of applications.gecko.id is not defined: 
https://github.com/SeleniumHQ/selenium/blob/3e99112a0d59878c6f31cc50fe8b0593c49e497f/javascript/node/selenium-webdriver/firefox.js#L172-L174